Sources/Regex/ASTTransform.swift

```diff
@@ -0,0 +1,33 @@
+extension AST {
+  public func withMatchLevel(_ level: CharacterClass.MatchLevel) -> AST {
```

Comment:
I think this might be ok for now, but ideally compilation would be parameterized over matching level rather than doing a bunch of tree transforms. Also makes it harder to update or refactor the AST.

Alternatively, we define walkers.

Reply:
Agreed. The change in this PR was just refactoring the original `RECode.withMatchLevel(_:)` to work for both the legacy and the matching engine backends.